core/core-spacemacs-buffer.el

```diff
@@ -261,15 +262,48 @@ If ALL is non-nil then truly all banners can be selected."
   "Return the full path to banner with index INDEX."
   (concat spacemacs-banner-directory (format "%03d-banner.txt" index)))
 
+(defun spacemacs-buffer//banner-fit-height-size ()
+  "Calculate height of startup banner to fit buffer contents.
+Returns height in units of line height with a minimum of 1."
+  ;; first determine number of lines occupied by startup list
+  (let* ((startup-list-line-height (apply #'+ (mapcar (lambda (l)
+                                                        ;; each section has 2
+                                                        ;; (header and
+                                                        ;; separator) plus
+                                                        ;; number of entries
+                                                        ;; number of lines
+                                                        (+ 2 (cdr l)))
+                                                      dotspacemacs-startup-lists)))
+         (mode)
+         ;; We determine the maximum available banner height by subtracting the
+         ;; number of lines in the home buffer contents (excl. logo and
+         ;; startup-list) and the number of lines in the startup list from the
+         ;; total available text lines
+         (image-height (- (window-text-height) 26 startup-list-line-height)))
```

Comment:
I think adding a proper constant name is more clear than a magic number of "26".

Reply:
It is quite hard to come up with a proper concise variable name for this. However, I have added the number to the comment to make the relation between the meaning of the comment and the code 'more explicit'.